drivers/serial/uart_stm32.c

```diff
@@ -1701,6 +1706,7 @@ static const struct uart_stm32_config uart_stm32_cfg_##index = {	\
 	.hw_flow_control = DT_INST_PROP(index, hw_flow_control),	\
 	.parity = DT_INST_ENUM_IDX_OR(index, parity, UART_CFG_PARITY_NONE),	\
 	.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(index),			\
+	.single_wire = DT_INST_PROP_OR(index, single_wire, false), \
```

Comment:
Similarly than `hw_flow_control`, booleans are `false` if not present.
So just use `DT_INST_PROP` instead of `DT_INST_PROP_OR`

Reply:
the problem here is, that this property might not be defined within the binding, so I guess the OR is needed, see https://github.com/zephyrproject-rtos/zephyr/pull/40525#discussion_r777373128